src/storage/aggregator.cc

```diff
@@ -143,48 +177,159 @@ bool Aggregator::Update(const std::string& key, const std::string& row, const ui
     return true;
 }
 
-bool Aggregator::GetAggrBuffer(const std::string& key, AggrBuffer* buffer) {
+bool Aggregator::FlushAll() {
     std::lock_guard<std::mutex> lock(mu_);
```

Comment:
oh. cannot use the global lock to lock and flushing. Blocking all the updates... 

It's not a good choice to do the flushing. let me think a better workaround

Reply:
we do the aggregator `FlushAll` only if there are binlog files deleted. This seems acceptable? 
Maybe in the future a unified interface can be used to restore tables and aggregators from binlog and snapshot.